filesystem/src/main/java/org/frankframework/filesystem/FileSystemActor.java

```diff
@@ -355,40 +355,38 @@ public Message doAction(@Nonnull Message input, ParameterValueList pvl, @Nonnull
 				}
 				case READDELETE: {
 					final F file = getFile(input, pvl);
-					try {
-						Message result = fileSystem.readFile(file, getCharset());
-						// Make a copy of a local file, otherwise the file is deleted after this method returns.
-						if (fileSystem instanceof LocalFileSystem) {
-							return result.copyMessage();
-						}
+					Message result = fileSystem.readFile(file, getCharset());
+					// Make a copy of a local file, otherwise the file is deleted after this method returns.
+					if (fileSystem instanceof LocalFileSystem) {
+						result = result.copyMessage();
+					} else {
 						result.preserve();
-						return result;
-					} finally {
-						fileSystem.deleteFile(file);
-						deleteEmptyFolder(file);
 					}
+					fileSystem.deleteFile(file);
```

Comment:
The order of execution is still the same, because the `finally` is done before the `return` actually happens.

However, there was a problem that an exception could happen while reading, and the `finally` would still remove the files. (And if there was also an exception when deleting files, you would not even see the original exception).

That is why I changed it this way.